app/models/dog_filter.rb

```diff
@@ -0,0 +1,55 @@
+#    Copyright 2017 Operation Paws for Homes
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+class DogFilter
+  attr_accessor :params, :manager
+
+  include Sortable
+
+  def initialize(params: {}, manager: false)
+    @params = params
+    @manager = manager
+  end
+
+  def filter
+    @dogs = Dog.filter(filtering_params)
+               .order( "#{sort_column} #{sort_direction}" )
+               .includes(:adoptions, :adopters, :comments, :primary_breed, :secondary_breed, :foster)
```

Comment:
hakiri flags as possible sql injection vulnerability line 27. However sort_column and sort_direction are both sanitized in the app/models/concerns/sortable.rb mixin.